github/plugin.py

```diff
@@ -735,3 +786,71 @@ def invite_user_to_gh_org_team(self, github_id, team_id):
+
+    def list_pull_requests(self, data, user_data):
+        """
+        Sets the default branch of a repo.
+
+        Command is as follows: !listprs <organization> <repo> <state>
+        :param data:
+        :return:
+        """
+        try:
+            parser = argparse.ArgumentParser()
+            parser.add_argument('org', type=str)
+            parser.add_argument('repo', type=str)
+            parser.add_argument('state', type=str)
+
+            args, unknown = parser.parse_known_args(args=preformat_args(data["text"]))
+            if len(unknown) > 0:
+                raise SystemExit()
+
+            args = vars(args)
+
+            # Check that we can use this org:
+            real_org = self.org_lookup[args["org"]][0]
+            reponame = extract_repo_name(args["repo"])
+            state = args["state"]
+
+        except KeyError as _:
+            send_error(data["channel"], '@{}: Invalid orgname sent in.  Run `!ListOrgs` to see the valid orgs.'
+                       .format(user_data["name"]), markdown=True)
+            return
+
+        except SystemExit as _:
+            send_info(data["channel"], "@{}: `!ListPRs` usage is:\n```!ListPRs <OrgThatHasRepo> "
+                                       "<Repo> <State>```\n"
+                                       "No special characters or spaces in the variables. \n"
+                                       "Run `!ListOrgs` to see the list of GitHub Organizations that I manage. "
+                      .format(user_data["name"]), markdown=True)
+            return
+
+        # Auth?
+        if self.commands["!ListPRs"].get("auth"):
+            if not self.commands["!ListPRs"]["auth"]["plugin"].authenticate(
+                    data, user_data, **self.commands["!ListPRs"]["auth"]["kwargs"]):
+                return
+
+        # Output that we are doing work:
+        send_info(data["channel"], "@{}: Working, Please wait...".format(user_data["name"]))
+
+        # Check that the repo exists:
+        repo_data = self.check_gh_for_existing_repo(reponame, real_org)
+        if not (repo_data):
+            send_error(data["channel"],
+                       "@{}: This repository does not exist in {}.".format(user_data["name"], real_org))
+            return False
+
+        # Grab all PRs [All states]
+        pull_requests = self.get_repo_prs(data, user_data, reponame, real_org, state)
+        if not (pull_requests):
+            return
+
+        headers = ["#PR", "Title", "Opened by", "Assignee", "State"]
+
+        rows = []
+        for pr in pull_requests:
+            assignee = pr['assignee']['login'] if pr['assignee'] is not None else '-'
+            rows.append([pr['number'],pr['title'], pr['user']['login'], assignee,pr['state'].title()])
+        # Done:
+        send_raw(data["channel"], text="Repository: *{}* \n\n```{}```".format(reponame, tabulate(rows, headers=headers, tablefmt='orgtbl')))
```

Comment:
I'm curious to see what this will look like.  I typically send messages as attachments so I can use Markdown for formatting. However, this is perfectly fine.